cookbook/basic_examples/log.py

```diff
@@ -0,0 +1,89 @@
+"""A basic example on how to log the data from a prompt and a chat completion."""
+import logging
+import os
+from typing import Optional
+
+import pandas as pd
+from sqlalchemy import JSON, Float, Integer, MetaData, String, create_engine
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column, sessionmaker
+
+from mirascope import OpenAIChat, Prompt, tags
+
+os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
+
+logger = logging.getLogger("mirascope")
+TABLE_NAME = "openai_chat_completions"
+
+
+class Base(DeclarativeBase):
+    pass
+
+
+class OpenAIChatCompletionTable(Base):
+    __tablename__ = TABLE_NAME
+    id: Mapped[int] = mapped_column(
+        Integer(), primary_key=True, autoincrement=True, nullable=False
+    )
+    template: Mapped[str] = mapped_column(String(), nullable=False)
+    inputs: Mapped[Optional[dict]] = mapped_column(JSONB)
+    tags: Mapped[Optional[list[str]]] = mapped_column(JSON)
+    start_time: Mapped[Optional[float]] = mapped_column(Float(), nullable=False)
+    end_time: Mapped[Optional[float]] = mapped_column(Float(), nullable=False)
+    output: Mapped[Optional[dict]] = mapped_column(JSONB)
+
+
+@tags(["recommendation_project"])
+class BookRecommendationPrompt(Prompt):
+    """
+    Can you recommend some books on {topic}?
+    """
+
+    topic: str
+
+
+USERNAME = "root"
+PASSWORD = ""
+HOST = "localhost"
+PORT = "5432"
+DB_NAME = "mirascope"
+engine = create_engine(f"postgresql://{USERNAME}:{PASSWORD}@{HOST}:{PORT}/{DB_NAME}")
+
+
+def create_database():
+    """Create the database and table for the OpenAI chat completions."""
+    metadata = MetaData()
+    table_objects = [Base.metadata.tables[TABLE_NAME]]
+    metadata.create_all(engine, tables=table_objects)
+
+
+def log_to_database(prompt_completion: dict):
```

Comment:
can we get some better typing here than just `dict`?

Reply:
what else would it be? do u want to create a pydantic model?

Reply:
Usually I'd expect `dict[str, Any]`